include/filedesc.h

```diff
@@ -0,0 +1,50 @@
+#ifndef __FILEDESC_H__
+#define __FILEDESC_H__
+#include <file.h>
+#include <bitstring.h>
+#include <mutex.h>
+
+typedef struct thread thread_t;
+
+typedef struct {
+  file_t **fdt_files; /* Open files array */
+  bitstr_t *fdt_map;  /* Bitmap of used fds */
```

Comment:
Please allocate statically, for simplicity, if you know its size is bounded and small.